.github/workflows/issue-write.yml

```diff
@@ -0,0 +1,128 @@
+name: Comment on an issue
+
+on:
+  workflow_run:
+    workflows: ["Check code formatting"]
+    types:
+      - completed
+
+permissions:
+  contents: read
+
+jobs:
+  pr-comment:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: >
+      github.event.workflow_run.event == 'pull_request'
+    steps:
+      - name: 'Download artifact'
+        uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+        with:
+          github-token: ${{ secrets.ISSUE_WRITE_DOWNLOAD_ARTIFACT }}
```

Comment:
@tstellar, we use clang-format check in downstream repository and this split "breaks" some functionality. According to my understanding, we should have `ISSUE_WRITE_DOWNLOAD_ARTIFACT` secret available to GitHub Actions in our repository in order to have comment from clang-format action. Could you clarify what permissions should be granted to `ISSUE_WRITE_DOWNLOAD_ARTIFACT` secret, please?
BTW, why can't we use `GITHUB_TOKEN` secret?

Reply:
The token needs `actions:read` permissions.  You can't use 'GITHUB_TOKEN`, because it's only scoped for the current workflow run.  See https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories

I do have a workaround that will allow us to download the artifact without a token, but that is part of a larger PR [here](https://github.com/llvm/llvm-project/pull/92058/files#diff-f79900b208fa205958090fbee62564a69295b9e5648172d62c211abe4a245abb).  If you think it would be useful to avoid using the token, I can try to pull out this change into its own PR.

Reply:
Thanks for clarifying!

> I do have a workaround that will allow us to download the artifact without a token, but that is part of a larger PR [here](https://github.com/llvm/llvm-project/pull/92058/files#diff-f79900b208fa205958090fbee62564a69295b9e5648172d62c211abe4a245abb). If you think it would be useful to avoid using the token, I can try to pull out this change into its own PR.

Do you plan to use "Unprivileged Download Artifact" in issue-write? If no, we will use the token. 

The only problem I have with introducing a new token is that it's kind of unexpected and requires some efforts to investigate why things are broken. If we can make it work out-of-box and keep the upstream project secure, it would be great.

Reply:
Thanks a lot for your help with this. I really appreciate the efforts to make GitHub Actions CI useful for downstream forks. 👍